src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,240 @@
+        catch (IOException | IllegalArgumentException ex) {
+            log(1, MSG_IO_EXCEPTION_KEY, file.getPath(), ex.getLocalizedMessage());
+        }
+
+        String previousProp = "";
+        int startLineNo = 0;
+
+        final Enumeration<Object> keys = properties.keys();
+
+        while (keys.hasMoreElements()) {
+
+            final String propKey = (String) keys.nextElement();
+
+            if (String.CASE_INSENSITIVE_ORDER.compare(previousProp, propKey) > 0) {
+
+                final int lineNo = getLineNumber(startLineNo, fileText, previousProp, propKey);
+                log(lineNo + 1, MSG_KEY, propKey, previousProp);
+                // start searching at position of the last reported validation
+                startLineNo = lineNo;
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns the index number where the key is detected (starting at 0).
+     * To assure that we get the correct line it starts at the point
+     * of the last occurrence.
+     * Also the previousProp should be in file before propKey.
+     *
+     * @param startLineNo start searching at line
+     * @param fileText {@link FileText} object contains the lines to process
+     * @param previousProp key name found last iteration, works only if valid
+     * @param propKey key name to look for
+     * @return index number of first occurrence. If no key found in properties file, 0 is returned
+     */
+    private static int getLineNumber(int startLineNo, FileText fileText,
+                                     String previousProp, String propKey) {
+
+        final int indexOfPreviousProp = getIndex(startLineNo, fileText, previousProp);
+        return getIndex(indexOfPreviousProp, fileText, propKey);
+    }
+
+    /**
+     * Inner method to get the index number of the position of keyName.
+     *
+     * @param startLineNo start searching at line
+     * @param fileText {@link FileText} object contains the lines to process
+     * @param keyName key name to look for
+     * @return index number of first occurrence. If no key found in properties file, 0 is returned
+     */
+    private static int getIndex(int startLineNo, FileText fileText, String keyName) {
+        final Pattern keyPattern = getKeyPattern(keyName);
+        int indexNumber = 0;
+        final Matcher matcher = keyPattern.matcher("");
+        for (int index = startLineNo; index < fileText.size(); index++) {
+            final String line = fileText.get(index);
+            matcher.reset(line);
+            if (matcher.matches()) {
+                indexNumber = index;
+                break;
+            }
+        }
+        return indexNumber;
+    }
+
+    /**
+     * Method returns regular expression pattern given key name.
+     *
+     * @param keyName
+     *            key name to look for
+     * @return regular expression pattern given key name
+     */
+    private static Pattern getKeyPattern(String keyName) {
+
```

Comment:
Please remove this blank line. We don't have a strict policy on this, but we want the whole code to be consistent. Therefore, all types must be sparse and all methods must be compact. Please reformat the code so that classes, interfaces and enumerations after `{` and before `}` have a blank line, but methods do not have blank lines at the beginning/end.

Reply:
ok. if it is your wish.